st2common/st2common/models/system/action.py

```diff
@@ -544,6 +544,15 @@ def _execute_remote_command(self, command):
 
         if output.failed:
             msg = 'Remote command %s failed.' % command
+            # XXX: Note Fabric doesn't handle unicode correctly if stdout or stderr contains
+            # unicode and action fails. For now, just log stdout and stderr so we can debug
+            # from logs.
+            # Fabric will show an exception traceback like:
+            # 'ascii' codec can't encode character u'\u2018' in position 93:
+            # ordinal not in range(128)
+            #
+            LOG.error('stderr: %s', output.stderr)
+            LOG.error('stdout: %s', output.stdout)
             LOG.error(msg)
             raise Exception(msg)
```

Comment:
Yeah, it's hard to explain here. If a task fails, none of the results there is collected cos this throws.
https://github.com/StackStorm/st2/blob/master/st2actions/st2actions/runners/fabricrunner.py#L144

That line throws because rm was setting the fabric state to fail even though the original fabric action to run the script succeeded. Am I making sense?